notification/slack/channel.go

```diff
@@ -310,7 +310,14 @@ func (s *ChannelSender) Send(ctx context.Context, msg notification.Message) (*no
 	vals.Set("channel", msg.Destination().Value)
 	switch t := msg.(type) {
 	case notification.Alert:
-		vals.Set("text", fmt.Sprintf("Alert: %s\n\n<%s>", t.Summary, cfg.CallbackURL("/alerts/"+strconv.Itoa(t.AlertID))))
+		if t.OriginalStatus != nil {
+			// escalated alert
+			vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID)
+			// updating text to simple text since the original alert will have all alert details, and this will be a thread reply message.
+			vals.Set("text", "Escalated.")
+		} else {
+			vals.Set("text", fmt.Sprintf("Alert: %s\n\n<%s>", t.Summary, cfg.CallbackURL("/alerts/"+strconv.Itoa(t.AlertID))))
+		}
```

Comment:
We should change this to a short-circuit rather than if/else, then leave the "default" behavior in the base scope.
```suggestion
		if t.OriginalStatus != nil {
			// Reply in thread if we already sent a message for this alert.
			vals.Set("thread_ts", t.OriginalStatus.ProviderMessageID)
			vals.Set("text", "Escalated.")
			break
		}

		vals.Set("text", fmt.Sprintf("Alert: %s\n\n<%s>", t.Summary, cfg.CallbackURL("/alerts/"+strconv.Itoa(t.AlertID))))
```